.github/workflows/CI.yml

```diff
@@ -31,3 +31,13 @@ jobs:
         run: rustup update ${{ matrix.rust }} && rustup default ${{ matrix.rust }}
       - name: Run tests
         run: cargo test --workspace
+
+  clippy_check:
+    runs-on: ubuntu-latest
+    # Make sure CI fails on all warnings, including Clippy lints
+    env:
+      RUSTFLAGS: "-Dwarnings"
+    steps:
+      - uses: actions/checkout@v3
+      - name: Run Clippy
+        run: cargo clippy --all-targets --all-features
```

Comment:
Could we make the clippy check only look for specific, conservative lints? I don't think I want to block builds on _all_ clippy warnings.